internal/pkg/cli/storage_init.go

```diff
@@ -292,6 +600,49 @@ func logicalIDSafe(input string) string {
 	return nonAlphaNum.ReplaceAllString(input, "")
 }
 
+func (o *initStorageOpts) generateDynamoDBConfig() (*dynamoDBConfig, error) {
+	cfg := &dynamoDBConfig{}
+	cfg.tableName = o.storageName
+	hashAttr, err := getAttrFromKey(o.partitionKey)
+	if err != nil {
+		return nil, err
+	}
+	rangeAttr, err := getAttrFromKey(o.sortKey)
+	if err != nil {
+		return nil, err
+	}
+	cfg.attributes = []attribute{
+		hashAttr,
+		rangeAttr,
+	}
+	if len(o.lsiSorts) != 0 {
+		for _, k := range o.lsiSorts {
+			lsiAttr, err := getAttrFromKey(k)
+			if err != nil {
+				return nil, err
+			}
+			cfg.attributes = append(cfg.attributes, lsiAttr)
+		}
+	}
+	cfg.partitionKey = hashAttr.name
+	cfg.sortKey = rangeAttr.name
+
+	return nil, nil
+}
+
+var regexpMatchAttribute = regexp.MustCompile("^(\\S+):([sbnSBN])")
+
+func getAttrFromKey(input string) (attribute, error) {
+	attrs := regexpMatchAttribute.FindStringSubmatch(input)
+	if len(attrs) == 0 {
+		return attribute{}, fmt.Errorf("parse attribute from key: %s", input)
+	}
+	return attribute{
+		name:        attrs[1],
```

Comment:
Is it possible to panic here because of out of index?

Reply:
No. Either the length will be 0 (no matches) and we'll return the error above, or 3, where the elements will be [full match, group 1,  group 2].